### PR TITLE
client: fix bad error handling in ll_lookup_inode

### PR DESCRIPTION
**Pull request by jtlayton**
It's not clear to me why we're bothering with the parent inode at all
in ll_lookup_inode. The caller only cares about the inode itself.

Just have the code call lookup_inode method directly and get rid of
ll_lookup_inode.

Signed-off-by: Jeff Layton <jlayton@redhat.com>

- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

### CODE REVIEW

**Comment by jtlayton** [comment]
This seems to work well enough, and it eliminates what seems to be a lot of extra work that appears to be unnecessary. Let me know if I'm missing something here though.

**Comment by jtlayton** [comment]
@gregsfortytwo mentioned that he thought all of the extra stuff is necessary to ensure that the inode in question is linked into the dentry tree.

I'm not sure either way, but the patch I just pushed may be safer. If we find that the inode already has a hashed dentry, just have _lookup_parent return -EINVAL, similar to how it works today with a root inode. We know that it's linked into the tree at that point, so we shouldn't need for the caller to do anything further.

**Comment by jtlayton** [comment]
I think I like these better. This moves most of the sanity checking of the inode into the caller, which makes for cleaner error handling.

File: src/client/Client.cc [hunk 1]
```diff
@@ -8591,8 +8590,6 @@ int Client::_lookup_parent(Inode *ino, const UserPerm& perms, Inode **parent)
       *parent = target.get();
       _ll_get(*parent);
       ldout(cct, 8) << __func__ << " found parent " << (*parent)->ino << dendl;
-    } else {
-      *parent = NULL;
```
Comment (gregsfortytwo) [inline src/client/Client.cc]: The rest of the PR looks fine but I generally prefer that we *do* set these values unconditionally. We've had a number of bugs not doing so in other parts of the code and I think we try to initialize them everywhere now.

Comment (jtlayton) [reply src/client/Client.cc]: I prefer leaving that sort of initialization to the caller, but I'll just drop that patch, as it's not critical here anyway.

**Comment by jtlayton** [comment]
Retest this please

**Comment by jtlayton** [comment]
Retest this please

**Comment by jtlayton** [comment]
Rebased onto current master, in the hopes that it'll fix the make check failures.

**Comment by jtlayton** [comment]
Hmm "make check" is still failing. I highly suspect that it has nothing to do with this PR, but I can't quite confirm that from the output in the build.

**Pull request merged**